### DISCUSSION TITLE
How to switch XCode version?

### DISCUSSION THREAD

**Discussion by Ghostbird** [dotnet/maui#12384] [Q&A] [answered]
When I try to build my MAUI app for iOS, it always fails for this reason:

> ILLINK : warning MT0079: The recommended Xcode version for Microsoft.iOS 16.1.1481 is XCode 14.1 or later. The current Xcode version (found in /Applications/Xcode_14.0.1.app/Contents/Developer) is 14.0.1. [/Users/runner/work/conneqt-app/conneqt-app/conneqt-app/Conneqt.csproj::TargetFramework=net7.0-ios]

No matter if I first `sudo xcode-select -s /Applications/Xcode_14.1.app` or set `DEVELOPER_DIR=/Applications/Xcode_14.1.app/Contents/Developer`, it will build using XCode 14.0.1 instead of XCode 14.1

Does anyone know how to get `dotnet` to use the correct XCode version?

Note in this example it's Github Actions, which is our normal CI pipeline, but we tried the same thing on a local mac mini and macbook.

Note 2: I do expect Github Actions to just start working on the January the 9th, because XCode 14.1 will become the default used at that point. However, that's just a happy coincidence, not an answer.

**Comment by matt-goldman** [comment, marked as answer]
Have you seen this? https://github.com/actions/runner-images/issues/6771

The suggested fix from @mattjohnsonpint worked for me.

**Reply by Ghostbird** [reply]
I'd completely forgotten that I asked this question. In the mean time, I had indeed found that issue and it solved it for me too.

**Reply by Ghostbird** [reply]
Also note that the January 9th update does not fix this issue. The reason has to do with the order in which the runner evaluates certain settings, combined with MAUI installation defaults, and is mentioned in the issue @matt-goldman mentions.

**Reply by Ghostbird** [reply]
For Github Actions ultimately I _only_ had to add:
```
env:
  MD_APPLE_SDK_ROOT: /Applications/Xcode_14.1.app
```
To the build step.
